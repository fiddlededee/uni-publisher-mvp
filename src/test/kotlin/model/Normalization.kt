package model

import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.dataformat.yaml.YAMLFactory
import com.fasterxml.jackson.dataformat.yaml.YAMLGenerator
import org.approvaltests.Approvals
import org.junit.jupiter.api.Test
import verify

class Normalization {

    @Test
    fun liAlreadyWrappedContents() {
        ListItem().apply { p { +"Некоторый текст"; span { +"и еще"; +" текст" } } }
            .wrapListItemContents().toYamlString().verify()
    }

    @Test
    fun liSimpleContents() {
        ListItem().apply { +"Некоторый текст"; span { +"и еще"; +" текст" } }
            .wrapListItemContents().toYamlString().verify()
    }
}

