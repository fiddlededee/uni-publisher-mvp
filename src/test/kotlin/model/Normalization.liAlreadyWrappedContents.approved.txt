!<ListItem>
children:
- !<Paragraph>
  children:
  - !<Text>
    text: "Некоторый текст"
  - !<Span>
    children:
    - !<Text>
      text: "и еще"
    - !<Text>
      text: " текст"
